public/app/core/logs_model.ts

```diff
@@ -176,14 +183,32 @@ export function makeSeriesForLogs(sortedRows: LogRowModel[], bucketSize: number,
       // for now setting the time step to be 0,
       // and handle the bar width by setting lineWidth instead of barWidth in flot options
       timeStep: 0,
+      highlightColor: tinycolor(series.color)
+        .darken(15)
+        .toString(),
     };
 
     return graphSeries;
   });
 }
 
-function isLogsData(series: DataFrame) {
-  return series.fields.some(f => f.type === FieldType.time) && series.fields.some(f => f.type === FieldType.string);
+// Checks if the data frame has at least one string field and one time field
```

Comment:
Same logic but re-written so only one iteration over series fields is done rather than two